src/main/scala/io/sdkman/vendor/release/routes/Validation.scala

```diff
@@ -45,4 +46,35 @@ trait Validation {
       17 >= version.length,
       ApiResponse(400, s"Version length exceeds 17 chars: $version").toJson.compactPrint
     )
+
+  def validateChecksumAlgorithms(checksums: Option[Map[String,String]]): Directive0 =
+    checksums
+      .map { checksum =>
+        val algorithms: Seq[String] = Seq(MD5.id, SHA1.id, SHA224.id, SHA256.id, SHA384.id, SHA512.id)
+
+        val invalidAlgorithm = checksum.keys.find { key =>
+            !algorithms.contains(key)
+        }.getOrElse("")
+
+        validate(
+          check = invalidAlgorithm.isEmpty,
+          ApiResponse(400, s"Invalid checksum algorithm: ${invalidAlgorithm}").toJson.compactPrint)
+
+      } getOrElse pass
+
+  def validateChecksums(checksums: Option[Map[String,String]]): Directive0 =
+    checksums
+      .map { checksum =>
+        var invalidChecksumAlgorithm: String = ""
+
+        for((k, v) <- checksum) {
+          if(v == null || v.isBlank) {
```

Comment:
This validation feels a bit naive, not sure if there is anything more that we can do. Would some regex perhaps help here?

Reply:
Ok, regex makes sense. I'll add that